src/fonduer/supervision/labeler.py

```diff
@@ -0,0 +1,237 @@
+import logging
+
+from fonduer.candidates.models import Candidate
+from fonduer.supervision.models import GoldLabelKey, Label, LabelKey
+from fonduer.utils.udf import UDF, UDFRunner
+from fonduer.utils.utils_udf import (
+    ALL_SPLITS,
+    add_keys,
+    batch_upsert_records,
+    get_cands_list_from_split,
+    get_docs_from_split,
+    get_mapping,
+    get_sparse_matrix,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def load_gold_labels(session, cand_lists, annotator_name="gold"):
+    """Load the sparse matrix for the specified annotator."""
+    return get_sparse_matrix(session, GoldLabelKey, cand_lists, key=annotator_name)
+
+
+class Labeler(UDFRunner):
+    """An operator to add Label Annotations to Candidates."""
+
+    def __init__(self, session, candidate_classes):
+        """Initialize the Labeler."""
+        super(Labeler, self).__init__(
+            session, LabelerUDF, candidate_classes=candidate_classes
+        )
+        self.candidate_classes = candidate_classes
+        self.lfs = []
+
+    def update(self, docs=None, split=0, lfs=None, **kwargs):
+        """Call apply with update=True."""
+        if lfs is None:
+            raise ValueError("Please provide a list of lists of labeling functions.")
+
+        if len(lfs) != len(self.candidate_classes):
+            raise ValueError("Please provide LFs for each candidate class.")
+
+        for i in range(len(self.lfs)):
+            # Filter out the new/updated LFs
+            self.lfs[i] = [
+                lf
+                for lf in self.lfs[i]
+                if lf.__name__ not in [_.__name__ for _ in lfs[i]]
+            ]
+            # Then add them
+            self.lfs[i].extend(lfs[i])
```

Comment:
Will this rerun all old lfs?

Reply:
Yes, for the candidate which need to be updated (e.g. if you change the definition of several of the old LFs, we want to reevaluate them.)

Reply:
Yes, this will. In general the number of LFs is small so it seem like this overhead is small relatively small as well as it just running some several python functions and ultimately still is only a single upsert in the DB. If we do not want to rerun old LFs, I can refactor this.

Reply:
I think we'd not rerun old/untouched/unchanged LFs so we don't need to refactor again.